flake.nix

```diff
@@ -467,9 +467,12 @@
             chmod u+w $out $out/plutarch.cabal
             # Remove stanzas from .cabal that won't work in GHC 8.10
             sed -i '/-- Everything below this line is deleted for GHC 8.10/,$d' $out/plutarch.cabal
-            # Remove packages that won't work in GHC 8.10 (yet)
+
+            # Prevent `sydtest-discover` from using GHC9 only modules when building with GHC810
+            # https://github.com/NorfairKing/sydtest/blob/master/sydtest-discover/src/Test/Syd/Discover.hs
             chmod -R u+w $out/plutarch-test
-            rm -rf $out/plutarch-test
+            rm -f $out/plutarch-test/src/Plutarch/MonadicSpec.hs
+            rm -f $out/plutarch-test/src/Plutarch/FieldSpec.hs
```

Comment:
@Norfairking FYI - this ^ is done to prevent sydtest-discover from using ghc9-only modules in ghc8. It would be great if it used the .cabal file to filter the modules.

Reply:
I don't think making `sydtest-discover` more complex like that will be a good tradeoff.
I think this is an okay hack for now, or we could use CPP